extensions/amp-mustache/0.1/amp-mustache.js

```diff
@@ -33,14 +36,37 @@ export class AmpMustache extends AMP.BaseTemplate {
 
   /** @override */
   compileCallback() {
+    /** @private @const {!JsonObject} */
+    this.nestedTemplates_ = dict();
+    const shadowTemplate = this.element.cloneNode(true);
+    let index = 0;
+    iterateCursor(templateContent(shadowTemplate).querySelectorAll('template'),
+        nestedTemplate => {
+          const nestedTemplateKey = `__AMP_NESTED_TEMPLATE_${index}`;
+          this.nestedTemplates_[nestedTemplateKey] =
+              nestedTemplate./*OK*/outerHTML;
+
+          const nestedTemplateAsVariable = this.element.ownerDocument
+              .createTextNode(`{{{${nestedTemplateKey}}}}`);
+          nestedTemplate.parentNode.replaceChild(nestedTemplateAsVariable,
```

Comment:
`DocumentFragment`s do not have a `#replaceChild`, making this fail if the template is a direct child of `templateContent`'s returning fragment.

Reply:
As you mentioned to me offline, it does have replaceChild as it inherits from Node